crates/algokit_transact/src/transactions/mod.rs

```diff
@@ -46,6 +46,56 @@ pub enum Transaction {
     // ApplicationCall(...),
 }
 
+pub struct FeeParams {
+    pub fee_per_byte: u64,
+    pub min_fee: u64,
+    pub extra_fee: Option<u64>,
+    pub max_fee: Option<u64>,
+}
+
+impl Transaction {
+    pub fn header(&self) -> &TransactionHeader {
+        match self {
+            Transaction::Payment(p) => &p.header,
+            Transaction::AssetTransfer(a) => &a.header,
+        }
+    }
+
+    pub fn header_mut(&mut self) -> &mut TransactionHeader {
+        match self {
+            Transaction::Payment(p) => &mut p.header,
+            Transaction::AssetTransfer(a) => &mut a.header,
+        }
+    }
+
+    pub fn assign_fee(&self, request: FeeParams) -> Result<Transaction, AlgoKitTransactError> {
+        let mut tx = self.clone();
+        let estimate_size = tx.estimate_size()?;
```

Comment:
```suggestion
        let estimated_size = tx.estimate_size()?;
```

Reply:
If the `fee_per_byte` is 0 we don't need to estimate the size of perform the calc.